src/client/opamListCommand.ml

```diff
@@ -370,12 +370,12 @@ let apply_selector ~base st = function
 
 let rec filter ~base st = function
   | Empty -> base
-  | Atom select -> apply_selector ~base st select
+  | Atom select -> base %% apply_selector ~base st select
```

Comment:
This line seems enough to warrant the post-condition that `filter ~base` ⊂ `base`, which would mean the `%%`s in the `And` and `Or` case could be removed ?

Reply:
Indeed. Good catch! Thanks!